lib/mayaUsd/ufe/UsdSetXformOpUndoableCommandBase.h

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include <mayaUsd/undo/UsdUndoableItem.h>
+
+#include <pxr/usd/usd/timeCode.h>
+
+#include <ufe/transform3dUndoableCommands.h>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+//! \brief Base class for TRS commands.
+//
+// Helper class to factor out common code for translate, rotate, scale
+// undoable commands.  It is templated on the type of the transform op.
+//
+// Developing commands to work with Maya TRS commands is made more difficult
```

Comment:
Commented on why we need a state machine: Maya never calls redo(), and has an undo() / set() call pair before manipulation begins that can trip up state capture for undo.